spark-operator/src/main/java/org/apache/spark/k8s/operator/config/SparkOperatorConf.java

```diff
@@ -0,0 +1,429 @@
+          .typeParameterClass(Integer.class)
+          .defaultValue(1)
+          .build();
+
+  public static final ConfigOption<Integer> RATE_LIMITER_REFRESH_PERIOD_SECONDS =
+      ConfigOption.<Integer>builder()
+          .key("spark.kubernetes.operator.rateLimiter.refreshPeriodSeconds")
+          .description("Operator rate limiter refresh period(in seconds) for each resource.")
+          .enableDynamicOverride(false)
+          .typeParameterClass(Integer.class)
+          .defaultValue(15)
+          .build();
+
+  public static final ConfigOption<Integer> RATE_LIMITER_LIMIT =
+      ConfigOption.<Integer>builder()
+          .key("spark.kubernetes.operator.rateLimiter.limit")
+          .description(
+              "Max number of reconcile loops triggered within the rate limiter refresh "
+                  + "period for each resource. Setting the limit <= 0 disables the "
+                  + "limiter.")
+          .enableDynamicOverride(false)
+          .typeParameterClass(Integer.class)
+          .defaultValue(5)
+          .build();
+
+  public static final ConfigOption<Integer> RETRY_INITIAL_INTERVAL_SECONDS =
+      ConfigOption.<Integer>builder()
+          .key("spark.kubernetes.operator.retry.initialIntervalSeconds")
+          .description("Initial interval(in seconds) of retries on unhandled controller errors.")
+          .enableDynamicOverride(false)
+          .typeParameterClass(Integer.class)
+          .defaultValue(5)
+          .build();
+
+  public static final ConfigOption<Double> RETRY_INTERVAL_MULTIPLIER =
+      ConfigOption.<Double>builder()
+          .key("spark.kubernetes.operator.retry.intervalMultiplier")
+          .description("Interval multiplier of retries on unhandled controller errors.")
+          .enableDynamicOverride(false)
+          .typeParameterClass(Double.class)
+          .defaultValue(1.5)
+          .build();
+
+  public static final ConfigOption<Integer> RETRY_MAX_INTERVAL_SECONDS =
+      ConfigOption.<Integer>builder()
+          .key("spark.kubernetes.operator.retry.maxIntervalSeconds")
+          .description(
+              "Max interval(in seconds) of retries on unhandled controller errors. "
+                  + "Set to -1 for unlimited.")
+          .enableDynamicOverride(false)
+          .typeParameterClass(Integer.class)
+          .defaultValue(-1)
+          .build();
+
+  public static final ConfigOption<Integer> RETRY_MAX_ATTEMPTS =
+      ConfigOption.<Integer>builder()
+          .key("spark.kubernetes.operator.retry.maxAttempts")
+          .description("Max attempts of retries on unhandled controller errors.")
+          .enableDynamicOverride(false)
+          .typeParameterClass(Integer.class)
+          .defaultValue(15)
+          .build();
+
+  public static final ConfigOption<Long> MAX_RETRY_ATTEMPT_ON_KUBE_SERVER_FAILURE =
+      ConfigOption.<Long>builder()
+          .key("spark.kubernetes.operator.retry.maxRetryAttemptOnKubeServerFailure")
```

Comment:
Just a question. Could you elaborate more about what kind of `Failure` do `Spark Operator` hit except `KubeServerFailure`?